File: sqrt.rb
#
# Adrian Statescu <mergesortv@gmail.com>
# This Method Computes SQuare RooT in Ruby.
# MIT License
#

def sqrt(n)

    an = n / 2.0

    anplus1 = ( an + (n / an).to_f ) / 2

    anEPS = 0.0001

    while (anplus1 - an).abs >= anEPS

       an = anplus1

       anplus1 = ( an + (n / an).to_f ) / 2

    end

   anplus1

end

puts "Enter n = "

n = gets.chomp

puts "sqrt(#{n}) = #{sqrt(n.to_i)}"

File: sqrt.go
/**
 *
 * Adrian Statescu (http://adrianstatescu.com) 
 *
 * Computes SQuar RooT (SQRT) using recurrence relations.
 *
 * anplus1 = ( an + n / an ) / 2 
 *
 * License MIT
 *
 */
package main

import ("fmt"
        "os"
        "strconv"
        "math")

func sqrt(n float64) float64 {    

     var an, anplus1, eps float64
     
     eps = 0.00001
  
     an =  n / 2.0

     anplus1 =  ( an + n / an) / 2.0 

     for math.Abs( anplus1 - an ) >= eps {

         an = anplus1
     
         anplus1 = ( an +  n / an) / 2.0 
     }

     return anplus1
}

func main() {

     var n float64

     n,_ = strconv.ParseFloat(os.Args[1], 64)

     fmt.Printf("%.10f", sqrt( n ))
}